Creation of host initiator group

gregoryb-at-ibm opened the pull request:
In this set of PRs we will allow users to create new host-initiator-groups on their storage systems through MIQ.

Part of https://github.com/ManageIQ/manageiq-providers-autosde/issues/108

Needs to be merged after https://github.com/ManageIQ/manageiq-api/pull/1105

![image](https://user-images.githubusercontent.com/68283004/145837479-0518a216-1a23-4b00-8500-1a0fdf22d9c9.png)

![image](https://user-images.githubusercontent.com/68283004/145837326-07a90cc2-5ef0-4713-94d4-4a8892b3c91c.png)

kavyanekkalapu commented:
@gregoryb-at-ibm Could you write testcases for the react file? You can find examples here -> https://github.com/ManageIQ/manageiq-ui-classic/tree/master/app/javascript/spec .

gregoryb-at-ibm commented:
> @gregoryb-at-ibm Could you write testcases for the react file? You can find examples here -> https://github.com/ManageIQ/manageiq-ui-classic/tree/master/app/javascript/spec .

On it, it's taking me a while to remember how :D

gregoryb-at-ibm commented:
Hi @kavyanekkalapu
I've added some specs. Could you see if it's sufficient coverage?

kavyanekkalapu commented:
> Hi @kavyanekkalapu I've added some specs. Could you see if it's sufficient coverage?

@gregoryb-at-ibm Yes looks great. I will review pr today.

kavyanekkalapu commented on app/javascript/spec/host-initiator-group-form/host-initiator-group.spec.js:
```diff
@@ -0,0 +1,131 @@
+import {act} from 'react-dom/test-utils';
+
+require('../helpers/miqSparkle.js');
+require('../helpers/miqAjaxButton.js');
```
@gregoryb-at-ibm I don't think we need any miqAjaxButton references here as we not using that in react file.

kavyanekkalapu commented on app/javascript/components/host-initiator-group-form/index.jsx:
```diff
@@ -0,0 +1,67 @@
+
+  const onSubmit = async(values) => {
+    miqSparkleOn();
+    const message = __('Host initiator group define');
```
Could you change message to  ` Add of Host initiator group "%s" has been successfully queued.`? %s is values.name. For examples you can check in other forms.

kavyanekkalapu commented on app/javascript/components/host-initiator-group-form/index.jsx:
```diff
@@ -0,0 +1,67 @@
+  };
+
+  const onCancel = () => {
+    const message = __('defining of host initiator was cancelled by the user');
```
Could you change message to  `Creation of new Host Aggregate was canceled by the user.`?

kavyanekkalapu commented on app/helpers/application_helper/toolbar/host_initiator_groups_center.rb:
```diff
@@ -19,6 +19,13 @@ class ApplicationHelper::Toolbar::HostInitiatorGroupsCenter < ApplicationHelper:
+          button(
+            :host_initiator_group_new,
+            'pficon pficon-add-circle-o fa-lg',
+            t = N_('Define a new host initiator group'),
```
Could you change this text to `Add new host initiator group` for consistency?

kavyanekkalapu commented on app/controllers/host_initiator_group_controller.rb:
```diff
@@ -15,6 +15,17 @@ def self.display_methods
+    if params[:storage_manager_id]
+      @storage_manager = find_record_with_rbac(ExtManagementSystem, params[:storage_manager_id])
+    end
+    drop_breadcrumb(:name => _("Define New %{table}") % {:table => ui_lookup(:table => table_name)},
```
Could you change drop breadcrumb to `Add new` instead of `Define New`?

kavyanekkalapu commented:
@gregoryb-at-ibm Could you fix testcases?

gregoryb-at-ibm commented:
> @gregoryb-at-ibm Could you fix testcases?

Fixed. Not sure how the snapshot got messed up.
But please don't merge before the API PR is merged.

kavyanekkalapu commented:
@gregoryb-at-ibm Your UI colors looks different. Which browser and OS  are you using? Can i access that machine. I am just curious on why colors/css looks different.

gregoryb-at-ibm commented:
> @gregoryb-at-ibm Your UI colors looks different. Which browser and OS are you using? Can i access that machine? I am just curious on why colors/css looks different.

Hi 
I usually access my MIQ dev machine through VNC. The VNC must have degraded the visuals.
Our dev machines are in a private network so unfortunately you won't be able to access them.
I'll try to build a new MIQ machine and take cleaner screenshots without VCN.

gregoryb-at-ibm commented:
I've replaced the screenshots with new screenshots taken directly from the browser.
I've also rebased my code on the master, and it caused the tests to start failing. I'm looking into that.

gregoryb-at-ibm commented:
Tests now pass and the related API PR is merged :)
